vars/buildPlugin.groovy

```diff
@@ -139,7 +139,29 @@ def call(Map params = [:]) {
                   mavenOptions += 'clean install'
                   try {
                     boolean usingArtifactCachingProxy = false
-                    if (artifactCachingProxyEnabled) {
+
+                    // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't wan't ACP
+                    boolean prLabelsContainSkipACP = false
+                    final String skipACPLabel = 'skip-artifact-caching-proxy'
+                    if (!env.BRANCH_IS_PRIMARY) {
+                      withCredentials([
+                        // This credential only exists on ci.jenkins.io and allows to check for GitHub PR labels
+                        // Would not be needed if GitHub Branch Sources plugin was be able to provide labels in an environment variable
+                        usernamePassword(credentialsId: 'app-ci.jenkins.io', usernameVariable: 'GITHUB_APP', passwordVariable: 'GH_TOKEN')
+                      ]) {
+                        // Creating the correct API URL to retrieve pull request labels from the $CHANGE_URL
+                        final String pullrequestLabelsApiURL = (env.CHANGE_URL).replace('/pull/', '/issues/').replace('/github.com/', '/api.github.com/repos/') + '/labels'
+                        if (isUnix()) {
+                          prLabelsContainSkipACP = sh(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" ' + pullrequestLabelsApiURL + ' | grep --ignore-case "skip-artifact-caching-proxy"', returnStatus: true) == 0
+                        } else {
+                          prLabelsContainSkipACP = bat(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer %GH_TOKEN%" ' + pullrequestLabelsApiURL + ' | findstr /i "skip-artifact-caching-proxy"', returnStdout: true) != ''
+                        }
+                      }
+                      if (prLabelsContainSkipACP) {
+                        echo "INFO: the label 'skip-artifact-caching-proxy' has been applied to the pull request, will use repo.jenkins-ci.org"
+                      }
```

Comment:
```suggestion
                          prLabels = sh(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" ' + pullrequestLabelsApiURL + '2>&1', returnStdout: true).trim()
                        } else {
                         prLabels = sh(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" ' + pullrequestLabelsApiURL + '2>&1', returnStdout: true).trim()
                        }
                      }
                      if (prLabels.contains('skip-artifact-caching-proxy')) {
                        echo "INFO: the label 'skip-artifact-caching-proxy' has been applied to the pull request, will use repo.jenkins-ci.org"
                      }

```

WDYT about a solution like this one? Since you already uses resources to retrieve shell step result in the pipeline, wether it's the stdout of a few line or a boolen does not change a lot of things.

Besides you could add a `try{} catch{}` block ardoun to avoid failing the pipeline if the step fails.

Additionally (or alternatively) you could use shell ternary form such as:

```
curl <....> | grep 'pattern to seach' || echo 'no label found'
```

so the command would never fail.

Reply:
I went back to my initial plan, with `returnStatut`, no issue anymore, same method for both Linux and Windows (I wasn't satisfied with the difference), no need for other changes.